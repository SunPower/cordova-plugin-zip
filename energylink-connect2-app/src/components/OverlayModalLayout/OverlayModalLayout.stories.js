import React from 'react'

import { storiesOf } from '@storybook/react'

import OverlayModalLayout from './index'

storiesOf('OverlayModalLayout', module).add('Simple', () => (
  <OverlayModalLayout />
))
