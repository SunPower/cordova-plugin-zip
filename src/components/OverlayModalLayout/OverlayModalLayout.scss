@import '../../styles/variables';

.overlay-modal-layout {
  position: fixed !important;
  bottom: 0;
  left: 0;
  width: 100% !important;
  margin: 0 !important;
  background-color: $gray-background-default;
  max-height: calc(100vh / 12 * 11) !important;
  overflow: hidden !important;

  &.is-half {
    height: 50vh;
    animation: slide-half 0.4s ease-out;
  }

  &.is-seven-eighths {
    height: calc(100vh / 8 * 7);
    animation: slide-seven-eighths 0.4s ease-out;
  }

  .modal-header {
    height: 5rem;
    border-bottom: 2px solid $gray-shadow-default;
    justify-content: center;
    align-items: center;

    h2 {
      font-size: $font-size-big;
      line-height: 1.1875rem;
      text-align: center;
      letter-spacing: 0.18em;
      color: $black;
      text-transform: uppercase;
    }
  }

  .modal-header {
    h2 {
      font-weight: $bold-text;
      font-size: $font-size;
      letter-spacing: $letter-spacing-default;
    }
  }
  .modal-body {
    flex-direction: column;
    .modal-select-options {
      width: 100%;

      .modal-option {
        cursor: pointer;

        width: 100%;
        height: 78px;
        align-items: center;
        flex-direction: row;

        &:not(:first-child) {
          border-top: 2px solid $gray-shadow-default;
        }
        span {
          font-size: $font-size-huge;
          line-height: 20px;
          letter-spacing: $letter-spacing-default;
          color: $black;

          &.text {
            flex-grow: 1;
          }

          &.text-left {
            font-weight: $bold-text;
          }
        }
      }
    }

    .backup-warning {
      span {
        font-size: $font-size;
        line-height: 19px;
        letter-spacing: $letter-spacing-default;
        color: $gray-text-alt2;

        &.highlighted {
          font-weight: $bold-text;
        }
      }
    }
  }
}

@keyframes slide-half {
  0% {
    height: 0;
  }
  100% {
    height: 50vh;
  }
}

@keyframes slide-seven-eighths {
  0% {
    height: 0;
  }
  100% {
    height: calc(100vh / 8 * 7);
  }
}
