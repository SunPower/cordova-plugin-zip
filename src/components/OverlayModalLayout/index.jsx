import React from 'react'
import { createPortal } from 'react-dom'
import { useDispatch, useSelector } from 'react-redux'
import clsx from 'clsx'
import { usePortal } from './usePortal'
import { TOGGLE_MODAL } from '../../state/actions/modal'

import './OverlayModalLayout.scss'

function closeModal(dispatch) {
  return () => dispatch(TOGGLE_MODAL({ isActive: false }))
}

const Modal = ({ id, header, className, children }) => {
  const dispatch = useDispatch()
  const modal = useSelector(state => state.global && state.global.modal)
  const { modalId, isActive } = modal
  const isModalActive = isActive && modalId === id
  const bodyClasses = clsx('modal-content', 'overlay-modal-layout', className)
  const modalClasses = clsx('modal', { 'is-active': isModalActive })
  return (
    <div className={modalClasses}>
      <div className="modal-background" onClick={closeModal(dispatch)}></div>
      <div className={bodyClasses}>
        <div className="modal-header is-flex">
          <h2>{header}</h2>
        </div>
        <div className="modal-body is-flex">{children}</div>
      </div>
    </div>
  )
}

const OverlayModalLayout = ({
  id = 'modal-root',
  className = 'is-half',
  header,
  children
}) => {
  const target = usePortal(id)
  return createPortal(
    <Modal id={id} header={header} className={className}>
      {children}
    </Modal>,
    target
  )
}

export default OverlayModalLayout
