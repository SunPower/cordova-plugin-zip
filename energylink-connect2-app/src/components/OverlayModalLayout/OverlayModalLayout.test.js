import React from 'react'
import { shallow } from 'enzyme'
import OverlayModalLayout from '.'
import * as reactRedux from 'react-redux'

describe('OverlayModalLayout component', () => {
  it('renders without crashing', () => {
    jest.spyOn(reactRedux, 'useSelector').mockImplementation(() => false)
    const component = shallow(<OverlayModalLayout />)
    expect(component).toMatchSnapshot()
  })
})
