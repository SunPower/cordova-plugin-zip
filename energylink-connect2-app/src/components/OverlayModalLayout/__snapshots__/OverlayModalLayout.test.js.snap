// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OverlayModalLayout component renders without crashing 1`] = `
<Portal
  containerInfo={<div />}
>
  <Modal
    className="is-half"
    id="modal-root"
  />
</Portal>
`;
